src/wazuh_modules/wm_ciscat.c

```diff
@@ -0,0 +1,530 @@
+        pthread_exit(NULL);
+    default:
+        mterror(WM_CISCAT_LOGTAG, "Unspecified content type for file '%s'. This shouldn't happen.", eval->path);
+        pthread_exit(NULL);
+    }
+
+    // Specify directory where saving reports
+
+    wm_strcat(&command, "-r", ' ');
+    wm_strcat(&command, WM_CISCAT_REPORTS, ' ');
+
+    // Set reports file name
+
+    wm_strcat(&command, "-rn", ' ');
+    wm_strcat(&command, "ciscat-report", ' ');
+
+    // Get xml reports
+
+    wm_strcat(&command, "-x", ' ');
+
+    // Do not create HTML report
+
+    wm_strcat(&command, "-n", ' ');
+
+    // Show all tests (Not selected included)
+
+    wm_strcat(&command, "-y", ' ');
+
+    // Send rootcheck message
+
+    snprintf(msg, OS_MAXSTR, "Starting CIS-CAT scan. File: %s. ", eval->path);
+    SendMSG(queue_fd, msg, "rootcheck", ROOTCHECK_MQ);
+
+    // Execute the scan
+
+    pid_t pid;
+
+    switch(pid = fork(), pid) {
+        case -1:
+            mterror(WM_CISCAT_LOGTAG, FORK_ERROR, errno, strerror(errno));
+            exit(1);
+        case 0:
+            // Child process
+            if (CreatePID("wazuh-modulesd:ciscat", getpid()) < 0)
+                mterror_exit(WM_CISCAT_LOGTAG, "Couldn't create PID file for child process: (%s)", strerror(errno));
+
+            if (chdir(path) < 0) {
+                mterror(WM_CISCAT_LOGTAG, "Unable to change working directory: %s", strerror(errno));
+                pthread_exit(NULL);
+            } else
+                mtdebug2(WM_CISCAT_LOGTAG, "Changing working directory to %s", path);
+
+            mtdebug1(WM_CISCAT_LOGTAG, "Launching command: %s", command);
+
+            switch (wm_exec(command, &output, &status, eval->timeout)) {
+                case 0:
+                    if (status > 0) {
+                        mtwarn(WM_CISCAT_LOGTAG, "Ignoring content '%s' due to error (%d).", eval->path, status);
+                        mtdebug2(WM_CISCAT_LOGTAG, "OUTPUT: %s", output);
+                        exit(1);
+                    }
+
+                    mtinfo(WM_CISCAT_LOGTAG, "Scan finished successfully. File: %s", eval->path);
+
+                    break;
+
+                case WM_ERROR_TIMEOUT:
+                    free(output);
+                    output = NULL;
+                    wm_strcat(&output, "ciscat: ERROR: Timeout expired.", '\0');
+                    mterror(WM_CISCAT_LOGTAG, "Timeout expired executing '%s'.", eval->path);
+                    break;
+
+                default:
+                    mterror(WM_CISCAT_LOGTAG, "Internal calling. Exiting...");
+                    exit(0);
+                    pthread_exit(NULL);
+            }
+
+            if (DeletePID("wazuh-modulesd:ciscat") < 0)
+                mterror_exit(WM_CISCAT_LOGTAG, "Couldn't delete PID file for child process: (%s)", strerror(errno));
+
+            _exit(0);
+
+        default:
+            // Parent process
+            switch(waitpid(-1, &child_status, 0)) {
```

Comment:
Are you sure you want to wait for **any** child process?
This implies that the wodle will continue when the script, or any child (like CIS-CAT or `xlstproc`) finishes.
I think that `waitpid(pid, &child_status, 0)` is better for you.